src/components/Select/components/SelectInner.vue

```diff
@@ -294,19 +294,23 @@ export default {
         return []
       }
 
-      return this.value.map(($v) => {
-        if (typeof $v === 'object') {
-          return $v
-        }
-
-        const option = this.options.find(($opt) => $opt[this.itemValue] === $v)
-
-        if (this.allowCreate && !option) {
-          return this.emitOption ? { [this.itemValue]: $v } : $v
-        }
-
-        return option
-      })
+      return this.value
+        .map(($v) => {
+          if (typeof $v === 'object') {
+            return $v
+          }
+
+          const option = this.options.find(
+            ($opt) => $opt[this.itemValue] === $v
+          )
+
+          if (this.allowCreate && !option) {
+            return this.emitOption ? { [this.itemValue]: $v } : $v
+          }
+
+          return option
+        })
+        .filter((option) => option !== null && option !== undefined)
```

Comment:
Hi @marcosh-ramos,
I know you just added the `.filter` method to the existing code for the fix. But here's a proposal to improve the code quality and make it easier to understand for everyone.

- This function has a lot of responsibilities, it could be split into multiple ones.
- From the `this.value` prop type I see that it can be of type `Boolean` `String` or `Array`. And this method uses `.map` on `this.value`. IMO, it would make sense to check the type first.
- The `$` prefix on the iteration value name is a debatable opinionated choice. At first, I thought it was from the 3rd party lib _"Vuelidate"_. Because when you use it, it adds `$v` to the Vue instance's prototype.
- The const name `option` is generic, it does not translate precisely what we are dealing with.
- I don't really get the difference between `this.value` and `this.itemValue` since the namings are not really self-explanatory 🙂 
- Extracting some conditions to accurately named constants can make the code clearer and easier to understand.

```suggestion
      if (!Array.isArray(this.value)) {
        // What should be returned in case `this.value` is not an array.
      }
      const values = this.value;
      return values
        .map((value) => {
        const isValueAnObject = typeof value === 'object' && value !== null;
          if (isValueAnObject) {
            return value
          }

          const selectedOption = this.options.find(
            (option) => option[this.itemValue] === value
          )
          
          const shouldCreateOption = this.allowCreate && !selectedOption

          if (shouldCreateOption) {
            return this.emitOption ? { [this.itemValue]: value } : value
          }

          return selectedOption
        })
        .filter((option) => option !== null && option !== undefined)
```

Reply:
Hi, @abenmassaoud thanks a million for the comments! about these points...

Reply:
<img width="743" alt="Screen Shot 2022-10-17 at 11 21 29" src="https://user-images.githubusercontent.com/7618411/196202136-a9d843e6-0eb5-45d8-b948-bfbed5884a0e.png">

There's an if at the beginning of the method `if (!this.hasValue || !this.multiple)` that verifies if the is select multiple, if not returns an empty array. So at this point(line 297)  `this.value` will be a list.

Reply:
<img width="777" alt="Screen Shot 2022-10-17 at 11 26 35" src="https://user-images.githubusercontent.com/7618411/196203350-ebc21e7a-f3f6-4800-b7d2-0b7e2fc86ba2.png">

I understand the point of a more descriptive name, but here don't see a lot of gain in creating this cont `values` since it will be used right after and only in that place.

Reply:
<img width="804" alt="Screen Shot 2022-10-17 at 11 30 46" src="https://user-images.githubusercontent.com/7618411/196204422-1b628221-9493-47aa-8041-abe149bc2b7a.png">

I saw that you add this `&& value !== null` to the verification, but I think that the old way of only validating if the value is an object and then returning is not wrong because we want this yearly exit if the `value` is equal `null`. There's no need to continue with the method execution if value is null.

Reply:
What do you think about these points?

Reply:
Hi @marcosh-ramos,

Thank you for your reply and sorry for the delay in response 🙂 
I have added 👍 reactions to the 2 points where I am aligned with your answers 😉 

Lastly, concerning the `const` creation, it was just to make the code more logical. Because the `this.value` naming is singular, and using `.map` on something singular did not seem that logical. But I get your point it's fine to forget about it 😉